FIX: ISXB-1153 fix exception on paste to empty actionmap list

smnwttbr opened the pull request:
### Description

Fixed an exception when pasting an action map into an empty action map list.  [ISXB-1150](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-1150)


### Testing status & QA

local package tests succeeded.

### Overall Product Risks

- Complexity:  low
- Halo Effect: small

### Comments to reviewers

_Please describe any additional information such as what to focus on, or historical info for the reviewers._

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - JIRA ticket linked, example ([case %<ID>%](https://issuetracker.unity3d.com/product/unity/issues/guid/<ID>)). If it is a private issue, just add the case ID without a link.
    - Jira port for the next release set as "Resolved".
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

After merge:

- [ ] Create forward/backward port if needed. If you are blocked from creating a forward port now please add a task to ISX-1444.

smnwttbr committed:
handle array length 0 correctly

smnwttbr committed:
update changlog

smnwttbr committed:
format fix

Pauliusd01 commented:
Getting an "Invalid action index to insert bindings before." error when pasting an action to an empty action list (does not occur on stable)

https://github.com/user-attachments/assets/5fb59a05-f8a2-49aa-a2c3-6e51a8c10d6a

Pauliusd01 reviewed (changes requested):
Just updating status

bmalrat committed:
FIX: Fixed unexpected control scheme switch when using `OnScreenControl` and pointer based schemes (ISXB-656) (#2023)

smnwttbr committed:
guard bad values in GetBindingIndexBeforeAction

Pauliusd01 reviewed (approved):
LGTM, found 2 more paste related bugs but they're not regressions so will file those separately. Checked: cut/copy pasting maps, actions and bindings between assets when they're empty, have assets, when the thing I'm cutting is the last on the list, when the actions/maps aren't saved yet in either asset (all with shortcuts and context menu). Also, undo/redo workflow and auto save behaviour.

AlexTyrer committed:
FIX: The "Add Control Scheme..." popup now requires explicit Save/Close user action (case ISXB-1131) (#2044)

* [Input System] The "Add Control Scheme..." popup now requires explicit Save/Close user action (case ISXB-1131)

o Previously clicking outside the popup window (anywhere in the editor) would close the popup window.

o This would leave a new blank (unnamed, orphaned) control scheme populating the UI - this was confusing.

o Now this popup will persist until the user explictly chooses to Save or Cancel.

* [Input System] Added CHANGELOG entry for "Add Control Scheme..." popup fix (case ISXB-1131)

* [Input System] Fix typo in CHANGELOG entry - removed extraneous final closing parenthesis.

* [Input System] Licensing failure - try limiting to version 2021.3.45f

smnwttbr committed:
Merge branch 'develop' into isxb-1153-paste-to-empty-maplist

smnwttbr committed:
Merge branch 'develop' into isxb-1153-paste-to-empty-maplist
